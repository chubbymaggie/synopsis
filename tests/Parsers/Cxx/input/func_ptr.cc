// Tests function pointers

typedef void PyObject;

// A function which takes a func ptr as a parameter
void insert(void* (*convert)(PyObject*), bool yesno);

// A function which takes a func ptr as a parameter
void insert(void* (*convert)(PyObject*, int), bool yesno);

// A function which returns a func ptr
void* (*insert2(int))(PyObject*);

int main()
{
  (void)insert2(1);
}
