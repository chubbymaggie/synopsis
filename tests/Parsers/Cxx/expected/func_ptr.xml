<?xml version='1.0' encoding='ISO-8859-1'?>
<?xml-stylesheet href='/home/stefan/synopsis/Synopsis/share/Synopsis/dump.css' type='text/css'?>
<ast>
 <declarations>
  <instance accessibility="0" class="Synopsis.AST.Typedef" constr="0" file="Parsers/Cxx/input/func_ptr.cc" language="C++" line="3" name="PyObject" type="typedef">
   <comments>
    <instance class="Synopsis.AST.Comment" line="0" suspect="1">
     <file>
      Parsers/Cxx/input/func_ptr.cc
     </file>
     <text>
      // Tests function pointers

     </text>
    </instance>
   </comments>
   <alias>
    <instance class="Synopsis.Type.Base" language="C++" name="void"/>
   </alias>
  </instance>
  <instance accessibility="0" class="Synopsis.AST.Function" file="Parsers/Cxx/input/func_ptr.cc" language="C++" line="6" name="insert(void*(*)(PyObject*),bool)" realname="insert" type="function">
   <comments>
    <instance class="Synopsis.AST.Comment" line="0" suspect="0">
     <file>
      Parsers/Cxx/input/func_ptr.cc
     </file>
     <text>
      // A function which takes a func ptr as a parameter

     </text>
    </instance>
   </comments>
   <parameters>
    <instance class="Synopsis.AST.Parameter" identifier="convert" value="">
     <type>
      <instance class="Synopsis.Type.Function" language="C++">
       <params>
        <instance class="Synopsis.Type.Modifier" language="C++">
         <alias>
          <instance class="Synopsis.Type.Declared" language="C++" name="PyObject">
           <declaration/>
          </instance>
         </alias>
         <postmod>
          *
         </postmod>
        </instance>
       </params>
       <premod>
        *
       </premod>
       <returnType>
        <instance class="Synopsis.Type.Modifier" language="C++">
         <alias/>
         <postmod>
          *
         </postmod>
        </instance>
       </returnType>
      </instance>
     </type>
    </instance>
    <instance class="Synopsis.AST.Parameter" identifier="yesno" value="">
     <type>
      <instance class="Synopsis.Type.Base" language="C++" name="bool"/>
     </type>
    </instance>
   </parameters>
   <returnType/>
  </instance>
  <instance accessibility="0" class="Synopsis.AST.Function" file="Parsers/Cxx/input/func_ptr.cc" language="C++" line="9" name="insert(void*(*)(PyObject*,int),bool)" realname="insert" type="function">
   <comments>
    <instance class="Synopsis.AST.Comment" line="0" suspect="0">
     <file>
      Parsers/Cxx/input/func_ptr.cc
     </file>
     <text>
      // A function which takes a func ptr as a parameter

     </text>
    </instance>
   </comments>
   <parameters>
    <instance class="Synopsis.AST.Parameter" identifier="convert" value="">
     <type>
      <instance class="Synopsis.Type.Function" language="C++">
       <params>
        <instance class="Synopsis.Type.Modifier" language="C++">
         <alias/>
         <postmod>
          *
         </postmod>
        </instance>
        <instance class="Synopsis.Type.Base" language="C++" name="int"/>
       </params>
       <premod>
        *
       </premod>
       <returnType>
        <instance class="Synopsis.Type.Modifier" language="C++">
         <alias/>
         <postmod>
          *
         </postmod>
        </instance>
       </returnType>
      </instance>
     </type>
    </instance>
    <instance class="Synopsis.AST.Parameter" identifier="yesno" value="">
     <type/>
    </instance>
   </parameters>
   <returnType/>
  </instance>
  <instance accessibility="0" class="Synopsis.AST.Function" file="Parsers/Cxx/input/func_ptr.cc" language="C++" line="12" name="insert2(int)" realname="insert2" type="function">
   <comments>
    <instance class="Synopsis.AST.Comment" line="0" suspect="0">
     <file>
      Parsers/Cxx/input/func_ptr.cc
     </file>
     <text>
      // A function which returns a func ptr

     </text>
    </instance>
   </comments>
   <parameters>
    <instance class="Synopsis.AST.Parameter" identifier="" value="">
     <type/>
    </instance>
   </parameters>
   <returnType>
    <instance class="Synopsis.Type.Function" language="C++">
     <params>
      <instance class="Synopsis.Type.Modifier" language="C++">
       <alias/>
       <postmod>
        *
       </postmod>
      </instance>
     </params>
     <premod>
      *
     </premod>
     <returnType>
      <instance class="Synopsis.Type.Modifier" language="C++">
       <alias/>
       <postmod>
        *
       </postmod>
      </instance>
     </returnType>
    </instance>
   </returnType>
  </instance>
  <instance accessibility="0" class="Synopsis.AST.Function" file="Parsers/Cxx/input/func_ptr.cc" language="C++" line="14" name="main()" realname="main" type="function">
   <returnType/>
  </instance>
 </declarations>
 <types>
  <instance class="Synopsis.Type.Declared" language="C++" name="insert(void*(*)(PyObject*,int),bool)">
   <declaration/>
  </instance>
  <instance class="Synopsis.Type.Declared" language="C++" name="insert2(int)">
   <declaration/>
  </instance>
  <instance class="Synopsis.Type.Declared" language="C++" name="main()">
   <declaration/>
  </instance>
  <instance class="Synopsis.Type.Declared" language="C++" name="insert(void*(*)(PyObject*),bool)">
   <declaration/>
  </instance>
 </types>
 <files>
  <file>
   Parsers/Cxx/input/func_ptr.cc
  </file>
 </files>
</ast>
